Index: Scribus/scribus/scribus.cpp
===================================================================
--- Scribus/scribus/scribus.cpp	(Revision 26264)
+++ Scribus/scribus/scribus.cpp	(Arbeitskopie)
@@ -3563,7 +3563,9 @@
 			if (doc->HasCMS)
 			{
 				recalcColors();
-				doc->RecalcPictures(&ScCore->InputProfiles, &ScCore->InputProfilesCMYK);
+				mainWindowProgressBar->setMaximum(doc->MasterItems.count() + doc->DocItems.count());
+				mainWindowProgressBar->setValue(0);
+				doc->RecalcPictures(&ScCore->InputProfiles, &ScCore->InputProfilesCMYK, mainWindowProgressBar);
 			}
 		}
 		else
